yarn-project/archiver/src/archiver/memory_archiver_store/memory_archiver_store.ts

```diff
@@ -703,34 +712,44 @@ export class MemoryArchiverStore implements ArchiverDataStore {
 
     const contractAddress = filter.contractAddress;
 
-    const logs: ExtendedUnencryptedL2Log[] = [];
+    const logs: ExtendedContractClassLog[] = [];
 
     for (; fromBlock < toBlock; fromBlock++) {
       const block = this.l2Blocks[fromBlock - INITIAL_L2_BLOCK_NUM];
       const blockLogs = this.contractClassLogsPerBlock.get(fromBlock);
 
       if (blockLogs) {
-        for (; txIndexInBlock < blockLogs.txLogs.length; txIndexInBlock++) {
-          const txLogs = blockLogs.txLogs[txIndexInBlock].unrollLogs();
-          for (; logIndexInTx < txLogs.length; logIndexInTx++) {
-            const log = txLogs[logIndexInTx];
-            if (
-              (!txHash || block.data.body.txEffects[txIndexInBlock].txHash.equals(txHash)) &&
-              (!contractAddress || log.contractAddress.equals(contractAddress))
-            ) {
-              logs.push(new ExtendedUnencryptedL2Log(new LogId(block.data.number, txIndexInBlock, logIndexInTx), log));
-              if (logs.length === this.maxLogs) {
-                return Promise.resolve({
-                  logs,
-                  maxLogsHit: true,
-                });
-              }
+        for (let logIndex = 0; logIndex < blockLogs.length; logIndex++) {
+          const log = blockLogs[logIndex];
+          // TODO(MW): this is a hack
+          const thisContractAddress = [
+            REGISTERER_CONTRACT_CLASS_REGISTERED_TAG,
+            REGISTERER_PRIVATE_FUNCTION_BROADCASTED_TAG,
+            REGISTERER_UNCONSTRAINED_FUNCTION_BROADCASTED_TAG,
+          ].includes(log.fields[0])
+            ? AztecAddress.fromNumber(REGISTERER_CONTRACT_ADDRESS)
+            : AztecAddress.ZERO;
```

Comment:
Similar to the comment on `log_store`, since we don't carry a `.contractAddress` around with the cc log, it's tricky once we get to the archiver to allow for filtering by address.